Fix truncate absence on android

Signed-off-by: Qi Zhou <atmgnd@outlook.com>
--- boost_1_61_0/libs/filesystem/src/operations.cpp	2016-05-06 05:11:33.000000000 +0800
+++ boost_1_61_0_new/libs/filesystem/src/operations-1.61.0.cpp	2020-05-09 23:41:22.803289162 +0800
@@ -220,7 +220,19 @@ typedef int err_t;
          || ::mkdir(to.c_str(),from_stat.st_mode)!= 0))
 #   define BOOST_COPY_FILE(F,T,FailIfExistsBool)copy_file_api(F, T, FailIfExistsBool)
 #   define BOOST_MOVE_FILE(OLD,NEW)(::rename(OLD, NEW)== 0)
+#if defined(__ANDROID__) || defined(ANDROID)
+static int BOOST_RESIZE_FILE(const char *path, off_t size)
+{
+	int result = -1;
+	int fd = open(path, O_WRONLY);
+	if (fd != -1)
+		result = ftruncate(fd, size);
+	close(fd);
+	return result;
+}
+#else
 #   define BOOST_RESIZE_FILE(P,SZ)(::truncate(P, SZ)== 0)
+#endif
 
 #   define BOOST_ERROR_NOT_SUPPORTED ENOSYS
 #   define BOOST_ERROR_ALREADY_EXISTS EEXIST
